--- a/drivers/misc/mediatek/base/power/udi_v2/mtk_udi.c
+++ b/drivers/misc/mediatek/base/power/udi_v2/mtk_udi.c
@@ -475,14 +475,14 @@ static ssize_t udi_jtag_clock_proc_write(struct file *file,
 	memset(DR_byte, 0, sizeof(DR_byte));
 
 	/* input check format */
-	if (sscanf(buf, "%4s %u %u %512s %u %u %512s %u",
+	if (sscanf(buf, "%4s %u %u %511s %u %u %511s %u",
 			&recv_key_word[0], &recv_buf[0],
 			&recv_buf[1], recv_char[0], &recv_buf[3],
 			&recv_buf[2], recv_char[1], &recv_buf[4]) == 8) {
 		/* 6 parameter */
 		IR_pause_count = recv_buf[3];
 		DR_pause_count = recv_buf[4];
-	} else if (sscanf(buf, "%4s %u %u %512s %u %512s",
+	} else if (sscanf(buf, "%4s %u %u %511s %u %511s",
 			&recv_key_word[0], &recv_buf[0],
 			&recv_buf[1], recv_char[0],
 			&recv_buf[2], recv_char[1]) == 6) {
